Modules/_io/textio.c

```diff
@@ -1719,21 +1719,31 @@ _io_TextIOWrapper_write_impl(textio *self, PyObject *text)
         bytes_len = PyBytes_GET_SIZE(b);
     }
 
-    if (self->pending_bytes == NULL) {
-        self->pending_bytes_count = 0;
-        self->pending_bytes = b;
-    }
-    else if (self->pending_bytes_count + bytes_len > self->chunk_size) {
+    PyObject *pending = self->pending_bytes;
+
+    if (self->pending_bytes_count + bytes_len > self->chunk_size) {
```

Comment:
How about `while` instead of `if`?

Reply:
That would also need a check to make sure that `bytes_len > self->chunk_size` isn't true.